jdat_notebooks/MRS_Mstar_analysis/JWST_Mstar_dataAnalysis_PointSourceDetectorBasedExtraction.ipynb

```diff
@@ -0,0 +1,2258 @@
+{
```

Comment:
<p>1. Please move the <code>tarfile</code> import to the top with the others.</p><p> </p><p>2. Gathering the names of these tar files into a list and looping through it would avoid the need to re-copy the filenames and similar code, which will help make future maintenance easier.</p><p> </p><p>Here is an approach you could use to condense this cell and the next one:</p><p><code></code><code></p><p>from pathlib import Path # would move to top with other imports</p><p> </p><p>tar_location = Path('.')</p><p>tar_files = ['file1.tar.gz', 'file2.tar.gz'] # and so on</p><p>tar_files_full = [tar_location / f for f in tar_files]</p><p>tar_urls = ['https://data.sceince.stsci.edu/file1',</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;'https://data.sceince.stsci.edu/file2'] # and so on</p><p> </p><p>for p, url in zip(tar_paths_full, tar_urls):</p><p>&nbsp;&nbsp;&nbsp;# check for un-tarred version of file</p><p>&nbsp;&nbsp;&nbsp;if os.path.exists(p.parent / p.stem):</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;print(f"{p.name} exists.")</p><p>&nbsp;&nbsp;&nbsp;else:</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# check for tarred version of file and un-tar</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if os.path.exists(p):</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;tar = tarfile.open(p, "r:gz")</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;tar.extractall()</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;tar.close()</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# otherwise, download and un-tar file</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;else:</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;urllib.request.urlretrieve(url, p)</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;print(f"Downloading {p.name}...")</p><p></code><code></code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/dat_pyinthesky/pull/162/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>